R/derive_vars_dy.R

```diff
@@ -0,0 +1,89 @@
+#' Derive Analysis Relative Day
+#'
+#' Adds the analysis relative day (`--DY`) to the dataset, i.e., study
+#' day of analysis date.
+#'
+#' @param dataset Input dataset
+#'
+#'   The columns specified by the `reference_date` and the `source_vars` parameter are
+#'   expected.
+#'
+#' @param reference_date The start date column, e.g., date of first treatment
+#'
+#'   A date or date-time object column is expected.
+#'
+#' @param source_vars A list of datetime or date variables created using `vars()` from
+#'   which dates are to be extracted. This can either be a list of date(time) variables
+#'   or named `--DY` variables and corresponding --DT(M) variables e.g.
+#'    vars(TRTSDTM, ASTDTM, AENDT) or vars(TRTSDT, ASTDTM, AENDT,DEATHDY=DTHDT)
+#'
+#' @author Teckla Akinyi
+#'
+#' @details The study day is derived as number of days from the reference date
+#'   to the end date. If it is nonnegative, one is added. I.e., the study day of the
+#'   reference date is 1. The input ---DT(M) is converted to ---DY
```

Comment:
"The input ---DT(M) is converted to ---DY" could be confusing as it is not clear that it referring to the name of the variables and not the values. What about updating to something like "If the name of the new variable is not specified explicitly, it is generated from the source variable name by replacing DT or DTM by DY."?

Reply:
agreed